docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,357 @@
+internal use only. Starburst recommends never using a self-signed cert for a
+production Trino server, especially one that has a globally valid DNS name and
+address.
+
+Certificate authorities are generally willing to sign CSRs for the server name
+``localhost`` or for an internal DNS name with a restricted
+`private network <https://en.wikipedia.org/wiki/Private_network>`_ address.
+
+However, CAs are generally unwilling to validate CSRs for self-signed certs
+under globally accessible circumstances. For a CA to validate a Certificate
+Signing Request for a server that has a globally accessible DNS name and URL,
+the CA must be able to validate that there is a server at that URL, and that
+your company owns the specified URL and DNS name. This requirement prevents
+signed certs that are valid on one site from being copied and re-used on another
+site. Modern browsers detect cases of valid certs used on the wrong server, and
+block attempts to connect to them.
+
+You cannot use a self-signed cert without validation from a Certificate
+Authority. Modern browsers detect servers running with such certs and either put
+up connection roadblocks or outright refuse to open such sites. Likewise, the
+Trino CLI detects a server running with an invalid self-signed cert and either
+warns against connecting or blocks the connection entirely.
+
+.. _tlsglossary:
+
+TLS Glossary
+------------
+
+.. _glossCA:
+
+CA
+    Certificate Authority, a trusted organization that scrupulously examines and
+    validates organizations and their proposed server URIs, and issues digital
+    certificates verified as valid for the requesting organization.
+
+.. _glossCert:
+
+Certificate
+    A public key `certificate
+    <https://en.wikipedia.org/wiki/Public_key_certificate>`_ issued by a CA,
+    sometimes abbreviated as *cert*, that verifies the ownership of a server's
+    keys. Certificates are specified in the `X.509
+    <https://en.wikipedia.org/wiki/X.509>`_ standard.
+
+.. _glossJKS:
+
+JKS
+    Java KeyStore, the system of public key cryptography supported as one part
+    of the Java security APIs. The legacy JKS system recognizes keys and
+    certificates stored in *keystore* files, typically with the ``.jks``
+    extension, and relies on a system-level list of CAs in *truststore* files
+    installed as part of the current Java installation.
+
+.. _glossKey:
+
+Key
+    A cryptographic key specified in
+    `PKCS #1 <https://en.wikipedia.org/wiki/PKCS_1>`_ as a pair of public and
+    private keys.
+
+.. _glossLB:
+
+Load Balancer
+    Software or a hardware device that sits on a network's outer edge or
+    firewall and accepts network connections on behalf of servers behind that
+    wall. Load balancers carefully manage network traffic, and can accept TLS
+    connections from incoming clients and pass those connections transparently
+    to servers behind the wall.
+
+.. _glossPEM:
+
+PEM
```

Comment:
Here, I disagree, because no one actually expands PEM. That is, no one ever writes "Privacy Enhanced Mail certificate" -- that would just cause head-scratching. By contrast, you can interchangeably write "Certificate Authority" and "CA" in a sentence. 

It was scholarly and nerdly of me to track down what PEM actually stands for, but in this case ... no one cares.